zanata-war/src/main/java/org/zanata/dao/ApplicationConfigurationDAO.java

```diff
@@ -1,12 +1,14 @@
 package org.zanata.dao;
 
 import org.hibernate.Session;
+
+import javax.enterprise.context.RequestScoped;
 import javax.inject.Named;
 import org.zanata.model.HApplicationConfiguration;
 
 @Named("applicationConfigurationDAO")
 
-@javax.enterprise.context.Dependent
+@RequestScoped
```

Comment:
Yes, I think @Dependent is best used only for classes which *have* to be
Dependent, like String - the lack of proxies is a subtle danger we don't
need to put up with for ordinary beans.

On 22 January 2016 at 10:17, Carlos Munoz <notifications@github.com> wrote:

> In
> zanata-war/src/main/java/org/zanata/dao/ApplicationConfigurationDAO.java
> <https://github.com/zanata/zanata-server/pull/1084#discussion_r50486499>:
>
> >  import javax.inject.Named;
> >  import org.zanata.model.HApplicationConfiguration;
> >
> >  @Named("applicationConfigurationDAO")
> >
> > -@javax.enterprise.context.Dependent
> > +@RequestScoped
>
> I think you're right... all other DAOs are request scoped. In light of
> this I remove my objection [image: :wink:]
>
> @seanf <https://github.com/seanf> @huangp <https://github.com/huangp> Do
> you remember why we had to make them Request scoped instead of dependent?
> Proxies perhaps?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1084/files#r50486499>.
>